<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>557</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>hehe</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="hlMain">
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="vbControl">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="vbControls">
         <property name="rightMargin">
          <number>1</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="Sliders">
           <item>
            <layout class="QVBoxLayout" name="Slider1">
             <item>
              <widget class="QLabel" name="labelFOV">
               <property name="text">
                <string>Field of View</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sbFOV">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsFOV">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="singleStep">
                <number>1</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="Slider2">
             <item>
              <widget class="QLabel" name="labelAngle">
               <property name="text">
                <string>Angle</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sbAngle">
               <property name="maximum">
                <number>360</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsAngle">
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QGroupBox" name="gbProjection_2">
           <property name="title">
            <string>Projection Mode</string>
           </property>
           <layout class="QVBoxLayout" name="gbProjection">
            <item>
             <widget class="QRadioButton" name="rbOrthogonal">
              <property name="text">
               <string>Orthogonal</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rbPerspective">
              <property name="text">
               <string>Perspective</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="verticalGroupBox_6">
           <property name="title">
            <string>Rotations</string>
           </property>
           <layout class="QVBoxLayout" name="gbRotation">
            <item>
             <layout class="QFormLayout" name="Roations">
              <item row="0" column="0">
               <widget class="QLabel" name="labelA">
                <property name="text">
                 <string>A/ Ambient</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSlider" name="hsRotationA">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelB">
                <property name="text">
                 <string>B/ Diffuse</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSlider" name="hsRotationB">
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="value">
                 <number>0</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="labelC">
                <property name="text">
                 <string>C/ Specular</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSlider" name="hsRotationC">
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="animationCB">
              <property name="text">
               <string>Animation</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="gimbalCB">
              <property name="text">
               <string>Gimbal Camera</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="gbClipping_2">
           <property name="title">
            <string>Clipping Plane</string>
           </property>
           <layout class="QVBoxLayout" name="gbClipping">
            <item>
             <layout class="QFormLayout" name="Clipping">
              <item row="1" column="0">
               <widget class="QLabel" name="labelFar">
                <property name="text">
                 <string>Far</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="labelNear">
                <property name="text">
                 <string>Near</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="dsbNear"/>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="dsbFar">
                <property name="value">
                 <double>2.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QFormLayout" name="flControlButtons">
           <item row="0" column="1">
            <widget class="QPushButton" name="rbReset">
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="MyGLWidget" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>480</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>vsFOV</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sbFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vsAngle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sbAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
